src/App.js

```diff
@@ -89,13 +91,15 @@ const InnerNav = ({ client, setClient }) => {
   const colors = getColors()
   const osReceiveState = useOsReceiveState()
   const osReceiveDispatch = useOsReceiveDispatch()
+  const filesToUpload = useFilesToUpload()
 
   return (
     <NativeIntentProvider
-      localMethods={localMethods(
-        client,
-        OsReceiveApi(client, osReceiveState, osReceiveDispatch)
-      )}
+      localMethods={localMethods(client, {
+        ...OsReceiveApi(client, osReceiveState, osReceiveDispatch),
+        hideSplashScreen: () =>
+          hideSplashScreen(undefined, filesToUpload.length > 0)
```

Comment:
Since this override the one form `/libs/intents/localMethods`, should we remove the `hideSplashScreen` it?

Reply:
I'm not sure, I think we should keep both to avoid potential race conditions. The first one would be a fallback if this one didn't override it yet.
I would need to test a bit more to be sure

Reply:
I'm not sure this scenario is possible as we inject this directly in the root component (app.js). But i'm ok with that, this should not block the approval